ball.js

```diff
@@ -0,0 +1,39 @@
+class Ball {
+  constructor(ctx, x, y, radius, color, speed) {
+    this.ctx = ctx;
+    this.x = x;
+    this.y = y;
+    this.radius = radius;
+    this.color = color;
+    this.speed = speed;
+    this.xDirection = 0;
+    this.yDirection = 1;
+  }
+
+  move() {
+    this.x += this.xDirection * this.speed;
+    this.y += this.yDirection * this.speed;
+    this.updateEdges();
+  }
+
+  updateEdges() {
+    this.top = this.y - this.radius;
+    this.bottom = this.y + this.radius;
+    this.left = this.x - this.radius;
+    this.right = this.x + this.radius;
+  }
+
+  setDirection(xDirection, yDirection) {
```

Comment:
why not make this 2 separate functions? you only need to flip a x/y direction when the ball hits a side from either left/right or top/bottom